bindings/rust/evmc-declare/src/lib.rs

```diff
@@ -337,40 +341,40 @@ fn build_execute_fn(names: &VMNameSet) -> proc_macro2::TokenStream {
         {
             use evmc_vm::EvmcVm;
 
-            let result = ::std::panic::catch_unwind(|| {
-                assert!(!instance.is_null());
-                // TODO: context is optional in case of the "precompiles" capability
-                assert!(!context.is_null());
-                assert!(!msg.is_null());
+            // TODO: context is optional in case of the "precompiles" capability
+            if instance.is_null() || context.is_null() || msg.is_null() || (code.is_null() && code_size != 0) {
+                // These are irrecoverable errors that violate the EVMC spec.
+                std::process::abort();
+            }
 
-                let execution_message: ::evmc_vm::ExecutionMessage = unsafe {
-                    msg.as_ref().expect("EVMC message is null").into()
-                };
+            assert!(!instance.is_null());
```

Comment:
The above conditional should handle all three of these asserts.

Reply:
I know, though I saw no harm keeping the asserts.